hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -261,6 +261,16 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted in call to rename", e);
+    } catch (RuntimeException e) {
+      // in case of table already exists,
+      // Hive rename operation throws exception as
+      // java.lang.RuntimeException:InvalidOperationException(message:new table <> already exists)
+      if (e.getCause() instanceof InvalidOperationException
+          && e.getCause().getMessage().contains(String.format("new table %s already exists", to))) {
```

Comment:
you might need to add a null check for the message as it's not guaranteed that an exception has a message